RELEASE/scripts/autoscend/auto_providers.ash

```diff
@@ -1192,3 +1192,229 @@ boolean provideMoxie(int amt, boolean doEquips)
+
+		if(pass())
+			return result();
+	}
+
+	void handleEffect(effect eff)
+	{
+		if(speculative)
+		{
+			delta += numeric_modifier(eff, "Item Drop");
+		}
+		auto_log_debug("We " + (speculative ? "can gain" : "just gained") + " " + eff.to_string() + ", now we have " + result());
+	}
+
+	boolean tryEffects(boolean [effect] effects)
+	{
+		foreach eff in effects
+		{
+			if(buffMaintain(eff, 0, 1, 1, speculative))
+				handleEffect(eff);
+			if(pass())
+				return true;
+		}
+		return false;
+	}
+
+	if(in_heavyrains())
+	{
+		buffMaintain($effect[Fishy Whiskers]); // HR only
+	}
+
+	// unlimited skills
+	if(tryEffects($effects[
+		Fat Leon\'s Phat Loot Lyric,
+		Singer\'s Faithful Ocelot
+	]))
+		return result();
+
+	if(canAsdonBuff($effect[Driving Observantly]))
+	{
+		if(!speculative)
+			asdonBuff($effect[Driving Observantly]);
+		handleEffect($effect[Driving Observantly]);
+	}
+	if(pass())
+		return result();
+
+	if(!bat_wantHowl(loc) && bat_formBats(speculative))
+	{
+		handleEffect($effect[Bats Form]);
+	}
+	if(pass())
+		return result();
+
+	if(auto_birdModifier("Item Drop") > 0)
+	{
+		if(tryEffects($effects[Blessing of the Bird]))
+			return result();
+	}
+
+	if(auto_favoriteBirdModifier("Item Drop") > 0)
+	{
+		if(tryEffects($effects[Blessing of Your Favorite Bird]))
+			return result();
+	}
+
+	// items
+	if(tryEffects($effects[
+		Joyful Resolve,
+		Fortunate Resolve,
+		Human-Human Hybrid,
+		Unusual Perspective,
+		Eagle Eyes,
+		Heart of Lavender,
+		Five Sticky Fingers
+	]))
+		return result();
+		
+	if(auto_sourceTerminalEnhanceLeft() > 0 && have_effect($effect[items.enh]) == 0 && auto_is_valid($effect[items.enh]))
+	{
+		if(!speculative)
+			auto_sourceTerminalEnhance("items");
+		handleEffect($effect[items.enh]);
+		if(pass())
+			return result();
+	}
+
+	//check specific item drop bonus
+	generic_t itemFoodNeed = zone_needItemFood(loc);
+	generic_t itemBoozeNeed = zone_needItemBooze(loc);
+	float itemDropFood = result() + simValue("Food Drop");
+	float itemDropBooze = result() + simValue("Booze Drop");
+	if(itemFoodNeed._boolean && itemDropFood < itemFoodNeed._float)
+	{
+		auto_log_debug("Trying food drop supplements");
+		//max at start of an expression with item and food drop is ineffective in combining them, have to let the maximizer try to add on top
+		addToMaximize("49food drop " + ceil(itemFoodNeed._float) + "max");
+		simMaximize();
+		itemDropFood = simValue("Item Drop") + simValue("Food Drop");
+	}
+	if(itemBoozeNeed._boolean && itemDropBooze < itemBoozeNeed._float)
+	{
+		auto_log_debug("Trying booze drop supplements");
+		addToMaximize("49booze drop " + ceil(itemBoozeNeed._float) + "max");
+		simMaximize();
+		itemDropBooze = simValue("Item Drop") + simValue("Booze Drop");
+		//no zone item yet needs both food and booze, bottle of Chateau de Vinegar exception is a cooking ingredient but doesn't use food drop bonus
+	}
+	if(pass())
+		return result();
+	
+
+	if(get_property("auto_dickstab").to_boolean())

```

Comment:
I would remove this clause. This property is a remnant of when Cheesecookie used cc_ascend as his personal looping script and wanted to extract value in-run. We should not care about such things. If we have the resources on hand in-run they are fair game to use for quest completion.

Reply:
It's nice to have these options so autoscend supports slow paths without using up all you saved in storage when it breaks ronin